compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/CompiledPackageSymbolEnter.java

```diff
@@ -249,9 +249,16 @@ private ConstantPoolEntry readCPEntry(DataInputStream dataInStream,
         UTF8CPEntry utf8CPEntry;
         switch (cpEntryType) {
             case CP_ENTRY_UTF8:
-                // Discard the length of bytes for now.
-                dataInStream.readShort();
-                return new UTF8CPEntry(dataInStream.readUTF());
+                short length = dataInStream.readShort();
+                String strValue = null;
+
+                // If the length of the bytes is -1, that means no UTF value has been written.
+                // i.e: string value represented by the UTF should be null.
+                // Therefore we read the UTF value only if the length >= 0.
+                if (length >= 0) {
```

Comment:
This means someone writing a null value. We should fix that.

Reply:
I believe this was added to allow null for string. We may be able to remove this check now.
I will have a look.